spec/unit/lib/cloud_controller/permissions_spec.rb

```diff
@@ -661,121 +666,103 @@ module VCAP::CloudController
-        it 'returns true for space developer' do
-          org.add_user(user)
-          space.add_developer(user)
-          expect(permissions.can_write_to_active_space?(space_guid)).to be true
-        end
-
-        it 'returns false for space manager' do
-          org.add_user(user)
-          space.add_manager(user)
-          expect(permissions.can_write_to_active_space?(space_guid)).to be false
+        before do
+          allow(membership).to receive(:has_any_roles?).with(Permissions::ROLES_FOR_SPACE_WRITING, space.id).and_return(true)
         end
-
-        it 'returns false for space auditor' do
+        it 'returns true' do
           org.add_user(user)
-          space.add_auditor(user)
-          expect(permissions.can_write_to_active_space?(space_guid)).to be false
-        end
-
-        it 'returns false for org manager' do
-          org.add_manager(user)
-          expect(permissions.can_write_to_active_space?(space_guid)).to be false
+          space.add_developer(user)
+          expect(permissions.can_write_to_active_space?(space.id)).to be true
         end
       end
     end
 
     describe '#can_manage_apps_in_active_space?' do
+      let(:membership) { instance_double(Membership) }
+
+      before do
+        allow(Membership).to receive(:new).with(user).and_return(membership)
+      end
       context 'user has no membership' do
+        before do
+          allow(membership).to receive(:has_any_roles?).with(Permissions::ROLES_FOR_APP_MANAGING, space.id).and_return(false)
+        end
         context 'and user is an admin' do
           it 'returns true' do
             set_current_user(user, { admin: true })
-            expect(permissions.can_manage_apps_in_active_space?(space_guid)).to be true
+            expect(permissions.can_manage_apps_in_active_space?(space.id)).to be true
           end
         end
 
         context 'and user is admin_read_only' do
           it 'return false' do
             set_current_user_as_admin_read_only
-            expect(permissions.can_manage_apps_in_active_space?(space_guid)).to be false
+            expect(permissions.can_manage_apps_in_active_space?(space.id)).to be false
           end
         end
 
         context 'and user is global auditor' do
           it 'return false' do
             set_current_user_as_global_auditor
-            expect(permissions.can_manage_apps_in_active_space?(space_guid)).to be false
+            expect(permissions.can_manage_apps_in_active_space?(space.id)).to be false
           end
         end
 
         context 'and user is not an admin' do
           it 'return false' do
             set_current_user(user)
-            expect(permissions.can_manage_apps_in_active_space?(space_guid)).to be false
+            expect(permissions.can_manage_apps_in_active_space?(space.id)).to be false
           end
         end
       end
 
       context 'user has valid membership' do
-        it 'returns true for space developer' do
```

Comment:
The previous checks for different roles also worked fine. Any reason why not to keep them?